docs/product-repo-plan-proposal.md

```diff
@@ -0,0 +1,56 @@
+# Enabling Performance Testing in the Product Repos
+
+## Value Statement
+
+We want performance testing to be consistent across all product repos that we test.
+All repos (in particular, coreclr and corefx) should run in the same way with the same benchmarks so that we can easily determine where a regression occured.
+
+## Promises
+
+- A consistent AzDO yaml template in arcade for performance testing that can be used across the product repositories with minimal set up for product repo owners.
+- A helix proj file that runs all of the benchmarks and splits them in a sane way
+
+## Success Metric(s)
+
+- Product repositories use the provided yaml template to run all performance testing
+- Product repositories do not try to skirt the yaml template to run performance testing only on the tests they care about
+- Setup time for adding performance testing is as simple as adding a new job that references the yaml template that we provide
+
+## Procedural
+
+- Combine the yaml we already use and, likely, the yaml that has been worked on in coreclr (why reinvent the wheel), with any additional logic necessary for pulling down and running the performance repository in product repos
+  - yaml will need to include any additional parameters/knobs that the product repos may want to turn (ie tiered/non-tiered/minopts, pgo/no-pgo, etc)
+  - yaml will need to include all uploading to benchview logic
```

Comment:
Why does the yaml need to contain logic for uploading results?  Would upload of results not take place from the helix machines that run the actual tests?

Reply:
Right now, upload is not integrated with the Helix API. The Benchview uploader is not compatible with the Helix API (I tried, it's not worth forcing them to be together), so until we replace the system, we still need to pass things like the Upload SAS token on private jobs, logic that we don't want to have to have the product repos know about. The fewer places we have to store that sort of logic, the fewer places we have to update when we do move to the new system, which we do plan on integrating with the helix API for uploading of data.

Reply:
Ok, so should I read this as, "Once we switch off of the Benchview uploader, we would move upload to be within the work that takes place on the helix machines"?